homeassistant/components/geniushub/__init__.py

```diff
@@ -31,33 +38,46 @@
 
 async def async_setup(hass, hass_config):
     """Create a Genius Hub system."""
-    from geniushubclient import GeniusHubClient  # noqa; pylint: disable=no-name-in-module
-
-    geniushub_data = hass.data[DOMAIN] = {}
-
     kwargs = dict(hass_config[DOMAIN])
     if CONF_HOST in kwargs:
         args = (kwargs.pop(CONF_HOST), )
     else:
         args = (kwargs.pop(CONF_TOKEN), )
 
+    hass.data[DOMAIN] = {}
+    data = hass.data[DOMAIN]['data'] = GeniusData(hass, args, kwargs)
     try:
-        client = geniushub_data['client'] = GeniusHubClient(
-            *args, **kwargs, session=async_get_clientsession(hass)
-        )
-
-        await client.hub.update()
-
+        await data._client.hub.update()  # pylint: disable=protected-access
     except AssertionError:  # assert response.status == HTTP_OK
         _LOGGER.warning(
-            "setup(): Failed, check your configuration.",
+            "Setup failed, check your configuration.",
             exc_info=True)
         return False
 
-    hass.async_create_task(async_load_platform(
-        hass, 'climate', DOMAIN, {}, hass_config))
+    async_track_time_interval(hass, data.async_update, SCAN_INTERVAL)
 
-    hass.async_create_task(async_load_platform(
-        hass, 'water_heater', DOMAIN, {}, hass_config))
+    for platform in ['climate', 'water_heater']:
+        hass.async_create_task(async_load_platform(
+            hass, platform, DOMAIN, {}, hass_config))
 
     return True
+
+
+class GeniusData:
+    """Container for geniushub client and data."""
+
+    def __init__(self, hass, args, kwargs):
+        """Initialize the geniushub client."""
+        self._hass = hass
+        self._session = async_get_clientsession(hass)
```

Comment:
The session is only used once. We don't need to store it as an instance attribute.

Reply:
Yeah - thought I'd taken that out - Done